python/cuml/dask/manifold/umap.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+
+from cuml.dask.common.input_utils import DistributedDataHandler
+from cuml.dask.common.base import DelayedTransformMixin
+from cuml.dask.manifold.base import BaseManifold
+
+from cuml.manifold.umap import UMAP as cumlUMAP
+
+
+class UMAP(BaseManifold,
+           DelayedTransformMixin):
+    def __init__(self, client=None, verbose=False, n_sampling=500, **kwargs):
+        super(UMAP, self).__init__(client, **kwargs)
+        self.local_model = cumlUMAP(verbose=verbose, MNMG=True, **kwargs)
```

Comment:
It seems strange to have the single GPU estimators aware of the multi-GPU estimators. It looks like this flag is just used to serialize the object for dask? I think a more maintainable solution could be to just create the `umap_params` lazily inside the fit and transform functions (and try to remove any places where the C++ side of those functions are modifying the values). AFAIK, there shouldn't be a need for explicit state in `fit()` to need to cross over to `transform()`.